### PR TITLE
Add firefox/sync (#9490, Fixes #9519, #9516)

### LINKED ISSUE #9519: Firefox Sync Coding
## Description

In https://github.com/mozilla/bedrock/issues/9516 we build a Sync page. This issue tracks coding.

- Url: /firefox/sync/
- Open for translation

---

#### Success Criteria
- [ ] Page is coded
- [ ] Redirect to /accounts/ is removed

### PR DESCRIPTION
**Pull request by amychurchwell**
## Description
Adds firefox/sync

## Issue / Bugzilla link
#9490
#9519
#9516

## Testing
[Copy](https://docs.google.com/document/d/1LE3SwAOvSajQ_3TC_lt87tj2Fxokxhe55BFiCngVuBA/edit)
[Design](https://github.com/mozilla/bedrock/issues/9518#issuecomment-706292804)

### CODE REVIEW

File: bedrock/firefox/templates/firefox/browsers/quantum.html
Comment (amychurchwell) [inline bedrock/firefox/templates/firefox/browsers/quantum.html:18]: unrelated to this PR, but I felt like the hero for the recent firefox/browsers/quantum page should be inside the `mzp-l-content` 🤷‍♀️

**Comment by amychurchwell** [comment]
Adding Do Not Merge until demo approved by Raphael & Chad

**Comment by amychurchwell** [comment]
Approved!

File: bedrock/firefox/templates/firefox/browsers/quantum.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/browsers/quantum.html:18]: The hero component already has an `mzp-l-content` container within it, so nesting it inside another one here creates some odd looking cropped content at smaller screen widths

Perhaps using one of the non-default [container theme classes](https://protocol.mozilla.org/patterns/templates/content-container.html), such as `mzp-t-content-xl` might look closer to how you want?

image

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:7]: Fluent IDs should use a standard prefix that associates them with the page where they are being used. This helps avoid naming collisions. See: https://bedrock.readthedocs.io/en/latest/l10n.html#ftl-string-ids

File: bedrock/firefox/templates/firefox/sync.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:18]: Same as for /quantum, the hero component should not be nested within another `mzp-l-content`

File: bedrock/firefox/templates/firefox/sync.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:28]: It feels a bit weird that this page is all about Sync, yet the primary CTA is still "Download Firefox" when I view the page in Firefox? At first i didn't notice the Sync instructions further down the page. Would it be worth replacing this with a sign-in button for Firefox users?

File: bedrock/firefox/templates/firefox/sync.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:65]: Should this link have some utms for tracking?

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:15]: Nit: curly `’`
```suggestion
we-store-your = We store your encrypted data on our servers so we can’t read it.
```

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:24]: Remove /en-US/ from link

File: bedrock/firefox/redirects.py
Comment (alexgibson) [inline bedrock/firefox/redirects.py:314]: This is now a 404 - did you mean to redirect it to /firefox/sync/?

**Review by alexgibson — APPROVED** [review]
A few nits / suggestions but overall the page is looking good 👍

**Review by alexgibson — CHANGES_REQUESTED** [review]
Oops, meant to request changes instead of approve. Trying again

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [reply l10n/en/firefox/sync.ftl:7]: I also realize now that we missed this with the /quantum page :(

File: bedrock/firefox/templates/firefox/sync.html
Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/sync.html:29]: Agreed! Done.

File: l10n/en/firefox/sync.ftl
Comment (amychurchwell) [reply l10n/en/firefox/sync.ftl]: PR to update firefox/quantum: https://github.com/mozilla/bedrock/pull/9623/files
thanks Alex :)

File: bedrock/firefox/redirects.py
Comment (alexgibson) [inline bedrock/firefox/redirects.py:314]:
```suggestion
    redirect(r'^sync/?$', 'firefox.sync'),
```

File: bedrock/firefox/templates/firefox/sync.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:76]: Remove `/en-US/`

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:11]: Should this be a verb instead brand name?
```suggestion
firefox-sync-sync-your-firefox = Sync your { -brand-name-firefox } on any device
```

File: l10n/en/firefox/sync.ftl
Comment (alexgibson) [inline l10n/en/firefox/sync.ftl:36]:
```suggestion
firefox-sync-learn-more = More help managing your { -brand-name-firefox-account }
```

File: bedrock/firefox/templates/firefox/sync.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/sync.html:31]: I think you still need to include the JS dependencies for the FxA button? https://bedrock.readthedocs.io/en/latest/firefox-accounts.html#cta-button-dependencies

**Review by alexgibson — CHANGES_REQUESTED** [review]
Looking good, just a few small fixups and the FxA button needs the JS libs for flow params.

**Review by alexgibson — APPROVED** [review]

**Pull request merged**
